### ISSUE TITLE
Feature, methods diffInOpenHours and diffInOpenMinutes

### ISSUE THREAD

**Issue by FuriosoJack** [spatie/opening-hours#139]
Hello.
Feachure, is posible diffInOpenHours and diffInOpenMinutes?

**Comment by kylekatarnls** [comment]
Hello, `spatie/opening-hours` aims to remain a low-level light library providing just the minimum stuff to work with open hours. But next/close methods should be enough to calculate diff with a loop like this:

```php
function floatDiffInOpenMinutes(DateTimeInterface $startDate, DateTimeInterface $endDate): float {
  $time = 0;

  if ($endDate < $startDate) {
    return -floatDiffInOpenMinutes($endDate, $startDate);
  }

  $date = $startDate->copy();

  while ($date < $endDate) {
    if ($date->isClosed()) {
      $date = $date->nextOpen();
      continue;
    }

    $nextDate = $date->nextClose();
    $time += min($endDate, $nextDate))->timestamp - $date->timestamp;
    $date = $nextDate;
  }

  return $time / 60;
}

function floatDiffInOpenHours(DateTimeInterface $startDate, DateTimeInterface $endDate): float {
  return floatDiffInOpenMinutes($startDate, $endDate) / 60;
}

function diffInOpenMinutes(DateTimeInterface $startDate, DateTimeInterface $endDate): int {
  return (int) floatDiffInOpenMinutes($startDate, $endDate);
}

function diffInOpenHours(DateTimeInterface $startDate, DateTimeInterface $endDate): int {
  return (int) floatDiffInOpenHours($startDate, $endDate);
}
```

Precision here is based on the DateTime timestamp (so microseconds are truncated at each adding but this should be enough in most situations).

**Comment by kylekatarnls** [comment]
Available in [2.8.0](https://github.com/spatie/opening-hours/releases/tag/2.8.0)
